scipy/special/specfun.c

```diff
@@ -0,0 +1,8418 @@
+    };
+    *id = 9;
+    // DLMF 13.4.4, integration up to C=12/X
+    a1 = a - 1.0;
+    b1 = b - a - 1.0;
+    c = 12.0 / x;
+    hu0 = 0.0;
+    for (m = 10; m < 101; m += 5) {
+        hu1 = 0.0;
+        g=0.5 * c / m;
+        d=g;
+        for (j = 1; j < (m + 1); j++) {
+            s = 0.0;
+                for (k = 1; k < 31; k++) {
+                    t1 = d + g * t[k-1];
+                    t2 = d - g * t[k-1];
+                    f1 = exp(-x*t1) * pow(t1, a1) * pow(1.0 + t1, b1);
+                    f2 = exp(-x*t2) * pow(t2, a1) * pow(1.0 + t2, b1);
+                    s += w[k-1]*(f1 + f2);
+                }
+            hu1 += s * g;
+            d += 2.0 * g;
+        }
+        if (fabs(1.0 - hu0/hu1) < 1.0e-9) { break; }
+        hu0 = hu1;
+    }
+    ga = specfun_gamma2(a);
+    hu1 /= ga;
+    // DLMF 13.4.4 with substitution t=C/(1-u)
+    // integration u from 0 to 1, i.e. t from C=12/X to infinity
+    for (m = 2; m < 11; m += 2) {
+        hu2 = 0.0;
+        g = 0.5 / m;
+        d = g;
+        for (j = 1; j < (m + 1); j++) {
+            s = 0.0;
+            for (k = 1; k < 31; k++) {
+                t1 = d + g * t[k-1];
+                t2 = d - g * t[k-1];
+                t3 = c / (1.0 - t1);
+                t4 = c / (1.0 - t2);
+                f1 = t3*t3 / c * exp(-x*t3)*pow(t3, a1)*pow(1.0 + t3, b1);
+                f2 = t4*t4 / c * exp(-x*t4)*pow(t4, a1)*pow(1.0 + t4, b1);
+                s += w[k-1]*(f1 + f2);
+            }
+            hu2 += s*g;
+            d += 2.0*g;
+        }
+        if (fabs(1.0 - hu0/hu2) < 1.0e-9) { break; }
+        hu0 = hu2;
+    }
+    ga = specfun_gamma2(a);
+    hu2 /= ga;
+    hu = hu1 + hu2;
+    return hu;
+}
+
+
+double specfun_chgul(double x, double a, double b, int *id) {
+
+    // =======================================================
+    // Purpose: Compute the confluent hypergeometric function
+    //          U(a,b,x) for large argument x
+    // Input  : a  --- Parameter
+    //          b  --- Parameter
+    //          x  --- Argument
+    // Output:  HU --- U(a,b,x)
+    //          ID --- Estimated number of significant digits
+    // =======================================================
+
+    int il1 = 0, il2 = 0, k, nm;
+    double aa, hu, r, r0 = 0.0, ra = 0.0;
+
+    *id = -100;
+    aa = a - b + 1.0;
+    if ((a == (int)a) && (a <= 0.0)) { il1 = 1; }
+    if ((aa == (int)aa) && (aa <= 0.0)) { il2 = 1; }
+    nm = 0;
+    if (il1) { nm = (int)fabs(a); }
+    if (il2) { nm = (int)fabs(aa); }
+    // IL1: DLMF 13.2.7 with k=-s-a
+    // IL2: DLMF 13.2.8
+    if (il1 || il2) {
+        hu = 1.0;
+        r = 1.0;
+        for (k = 1; k < (nm + 1); k++) {
+            r *= -(a + k - 1.0)*(a - b + k) / (k*x);
+            hu += r;
+        }
+        hu *= pow(x, -a);
+        *id = 10;
+    } else {
+        // DLMF 13.7.3
+        hu = 1.0;
+        r = 1.0;
+        for (k = 1; k < (nm + 1); k++) {
```

Comment:
Found it, this loop should run up to 25. That seems to be the problem and why only aarch64 fails I have no idea.